app/projectsmodel.h

```diff
@@ -116,17 +120,21 @@ class ProjectsModel : public QAbstractListModel
     //! lists projects, either fetch more or get first, search expression
     Q_INVOKABLE void listProjects( const QString &searchExpression = QString(), int page = 1 );
 
-    //! lists projects via listProjectsByName API, used in LocalProjectsModel
-    Q_INVOKABLE void listProjectsByName();
+    /**
+     * Lists projects via listProjectsByName API, used in LocalProjectsModel. If projectIds is empty fetches all local
+     * projects else fetches only specified projects.
+     * \param projectIds List of project IDs to fetch, by default empty
+     */
+    Q_INVOKABLE void fetchProjectsByProjectId( const QStringList &projectIds = QStringList() );
 
     //! Syncs specified project - upload or update
     Q_INVOKABLE void syncProject( const QString &projectId );
 
     //! Stops running project upload or update
-    Q_INVOKABLE void stopProjectSync( const QString &projectId );
+    Q_INVOKABLE void stopProjectSync( const QString &projectId ) const;
```

Comment:
This one should not be `const`, it alters things (maybe not directly, but it does)!